### PR TITLE
feat: add new provide leptonAI

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add new model provider LeptonAI

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](/CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: inference/providers/leptonai/chat_completion.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+from typing import Tuple, Dict
+from provider_dependency.chat_completion import *
+from .utils import *
+
+logger = logging.getLogger(__name__)
+
+
+model_id_list = [
```
Comment (jameszyao) [inline inference/providers/leptonai/chat_completion.py:8]: Remove constant provider_model_id

File: inference/providers/leptonai/chat_completion.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        stream: bool,
+        provider_model_id: str,
+        messages: List[ChatCompletionMessage],
+        credentials: ProviderCredentials,
+        configs: ChatCompletionModelConfiguration,
+        function_call: Optional[str] = None,
+        functions: Optional[List[ChatCompletionFunction]] = None,
+    ) -> Tuple[str, Dict, Dict]:
+        # todo accept user's api_url, the resulting URL will be a specific endpoint for a Lepton AI chat model
+        if provider_model_id not in model_id_list:
```
Comment (jameszyao) [inline inference/providers/leptonai/chat_completion.py:106]: Directly pass provider_model_id to the provider. Raise provider error if the model_id not found

**Review by jameszyao — APPROVED** [review]
LGTM

**Review by SimsonW — DISMISSED** [review]
LGTM

**Review by SimsonW — APPROVED** [review]
LGTM

**Review by jameszyao — APPROVED** [review]
LGTM

**Pull request merged**